Automattic/jetpack
# [Newsletter] Create default membership product silently in newsletter context

Fixes https://github.com/Automattic/gold/issues/49

## Proposed changes:

* Add a separate selector for the Newsletter products that resolves and silences the creation of the default product  

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Create a new site
* Sandbox the site
* Go to "new post" page and open the newsletter 
* **You should not see** any snackbar saying "Successfully created payment plan"
<img width="666" alt="Screenshot 2023-05-25 at 15 34 07" src="https://github.com/Automattic/jetpack/assets/790558/c4a8e6aa-0368-4ce5-989b-966912776680">

* Jest tests:
```
cd projects/plugins/jetpack && pnpm test-extensions extensions/store/membership-products/test
```
I tried to implement some resolver tests, but I failed to make them work as expected....
